File: usecase-ensured/src/main/java/com/github/usecase_ensured/internal/Request.java
package com.github.usecase_ensured.internal;

import com.fasterxml.jackson.databind.JsonNode;
import io.restassured.http.Headers;

public abstract class Request {
    protected final JsonNode request;

    protected Request(JsonNode request) {
        this.request = request;
    }

    public JsonNode request() {
        return request;
    }

    public abstract String url();
    public abstract String body();
    public abstract Headers headers();
    public abstract String method();

    public enum Method {
        GET, POST, PUT, DELETE
    }
}

File: lib/src/main/kotlin/com/github/bitknot_project/progressive_testing/data/Request.kt
package com.github.bitknot_project.progressive_testing.data

import io.restassured.http.Header

internal data class Request(
    var method: Method,
    var headers: List<Header>,
    var url: String,
    var body: String?
) {

    internal enum class Method {
        GET,
        POST,
        PUT,
        DELETE
    }
}
